docker/scripts/cyber_start.sh

```diff
@@ -15,265 +15,509 @@
+        eval "${qemu_cmd}" >/dev/null
+        info "Multiarch support has been enabled. Ref: ${refer}"
+    else
+        warning "Multiarch support hasn't been enabled. Please make sure the"
+        warning "following command has been executed before continuing:"
+        warning "  ${qemu_cmd}"
+        warning "Refer to ${refer} for more."
+        exit 1
+    fi
+}
+
+function start_cyber_container() {
+    local image="$1"
+    # if [ ! -d "${CACHE_ROOT_DIR}" ]; then
+    #    mkdir "${CACHE_ROOT_DIR}"
+    # fi
+    info "Starting docker container \"${CYBER_CONTAINER}\" ..."
+
+    local user="${USER}"
+    local uid="$(id -u)"
+
+    local group=$(id -g -n)
+    local gid=$(id -g)
+    local local_host="$(hostname)"
 
-    ${DOCKER_CMD} run -it \
+    local local_volumes
+    setup_devices_and_mount_volumes local_volumes
+
+    local display="$(determine_display)"
+
+    set -x
+    ${DOCKER_RUN_CMD} -it \
         -d \
         --privileged \
-        --name $APOLLO_CYBER \
-        -e DISPLAY=$display \
-        -e DOCKER_USER=$USER \
-        -e USER=$USER \
-        -e DOCKER_USER_ID=$USER_ID \
-        -e DOCKER_GRP="$GRP" \
-        -e DOCKER_GRP_ID=$GRP_ID \
-        -e DOCKER_IMG=$IMG \
-        -e USE_GPU=$USE_GPU \
+        --name "${CYBER_CONTAINER}" \
+        -e DISPLAY="${display}" \
+        -e DOCKER_USER="${user}" \
+        -e USER="${user}" \
+        -e DOCKER_USER_ID="${uid}" \
+        -e DOCKER_GRP="${group}" \
+        -e DOCKER_GRP_ID="${gid}" \
+        -e DOCKER_IMG="${image}" \
+        -e USE_GPU="${USE_GPU}" \
+        -e NVIDIA_VISIBLE_DEVICES=all \
+        -e NVIDIA_DRIVER_CAPABILITIES=compute,video,graphics,utility \
         -e OMP_NUM_THREADS=1 \
-        $(local_volumes) \
+        ${local_volumes} \
         --net host \
         -w /apollo \
-        --add-host in_cyber_docker:127.0.0.1 \
-        --add-host ${LOCAL_HOST}:127.0.0.1 \
-        --hostname in_cyber_docker \
+        --add-host "${CYBER_INSIDE}:172.0.0.1" \
+        --add-host "${local_host}:127.0.0.1" \
+        --hostname "${CYBER_INSIDE}" \
         --shm-size 2G \
         --pid=host \
-        -v /dev/null:/dev/raw1394 \
-        $IMG \
+        "${image}" \
         /bin/bash
 
     if [ $? -ne 0 ];then
-        error "Failed to start docker container \"${APOLLO_CYBER}\" based on image: $IMG"
+        error "Failed to start docker container \"${CYBER_CONTAINER}\" based on image: ${image}"
         exit 1
     fi
+    set +x
+}
 
-    if [ ${ARCH} == "x86_64" ]; then
-        if [ "${USER}" != "root" ]; then
-            docker exec -u root $APOLLO_CYBER bash -c '/apollo/scripts/docker_start_user.sh'
-        fi
-    else
-        warning "!!! Due to the problem with 'docker exec' on Drive PX platform, please run '/apollo/scripts/docker_start_user.sh' for the first time when you get into the docker !!!"
+function after_run_setup() {
+    if [[ "${USER}" != "root" ]]; then
```

Comment:
As we removed install_user when docker build, `docker_start_user.sh` have to be exectuted when we run `cyber_start.sh`. Tested.